controllers/datamovement_controller_test.go

```diff
@@ -91,7 +197,288 @@ var _ = Describe("Data Movement Test" /*Ordered, (Ginkgo v2)*/, func() {
 		})
 	})
 
+	Context("when the dm configmap has specified a dmCommand", func() {
+		BeforeEach(func() {
+			setDefaultConfigValues()
+			cmData[DMConfigKeyCmd] = "ls -l"
+		})
+		It("should use that command instead of the default mpirun", func() {
+			Eventually(func(g Gomega) string {
+				cmd := ""
+				g.Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(dm), dm)).To(Succeed())
+				if dm.Status.CommandStatus != nil {
+					cmd = dm.Status.CommandStatus.Command
+				}
+				return cmd
+			}).Should(Equal("/bin/" + cmData[DMConfigKeyCmd]))
+		})
+	})
+
+	Context("when the dm config map has specified a valid dmNumProcesses", func() {
+		BeforeEach(func() {
+			setDefaultConfigValues()
+			cmData[DMConfigKeyCmd] = ""
+			cmData[DMConfigKeyNumProcesses] = "17"
+		})
+
+		It("should use that number for the -np flag", func() {
+			Eventually(func(g Gomega) string {
+				cmd := ""
+				g.Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(dm), dm)).To(Succeed())
+				if dm.Status.CommandStatus != nil {
+					cmd = dm.Status.CommandStatus.Command
+				}
+				return cmd
+			}).Should(Equal(fmt.Sprintf(
+				"mpirun --allow-run-as-root -np 17 --host localhost dcp --progress 1 %s %s", srcPath, destPath)))
+		})
+	})
+
+	Context("when the dm config map has specified a invalid dmNumProcesses", func() {
+		BeforeEach(func() {
+			setDefaultConfigValues()
+			cmData[DMConfigKeyCmd] = ""
+			cmData[DMConfigKeyNumProcesses] = "aa"
+		})
+
+		It("should use the default number for -np flag", func() {
+			Eventually(func(g Gomega) string {
+				cmd := ""
+				g.Expect(k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(dm), dm)).To(Succeed())
+				if dm.Status.CommandStatus != nil {
+					cmd = dm.Status.CommandStatus.Command
+				}
+				return cmd
+			}).Should(Equal(fmt.Sprintf(
+				"mpirun --allow-run-as-root -np 1 --host localhost dcp --progress 1 %s %s", srcPath, destPath)))
+		})
+	})
+
+	Context("when the dm config map has specified a valid dmProgressInterval", func() {
+		BeforeEach(func() {
+			setDefaultConfigValues()
+			cmData[DMConfigKeyProgInterval] = "25s"
+		})
+
+		It("parseConfigMapValues should return that value", func() {
+			configMap := &corev1.ConfigMap{}
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: DMConfigMapName, Namespace: DMConfigMapNamespace}, configMap)).To(Succeed())
+			Expect(configMap.GetLabels()[testLabelKey]).To(Equal(testLabel))
```

Comment:
JustBeforeEach already checks the label is current, I don't see why this is needed

Reply:
Just me trying to increase the stability of these test since I was having issues.

Reply:
I removed these also